Core/Core/Transports/Memory.cs

```diff
@@ -57,10 +57,7 @@ public void BeginWrite()
   public void SaveObject(string id, string serializedObject)
   {
     var stopwatch = Stopwatch.StartNew();
-    if (CancellationToken.IsCancellationRequested)
-    {
-      return; // Check for cancellation
-    }
+    CancellationToken.ThrowIfCancellationRequested();
```

Comment:
Kind of nervous about the frequency of these checks - typically cancellation checks are performmed within a loop, I don't think it has to be caught absolutely immediately... but again, it's a little outside the remit